src/workerd/io/actor-cache.h

```diff
@@ -407,11 +407,21 @@ class ActorCache final: public ActorCacheInterface {
     // before the read completes.
     const Value value;
     EntryValueStatus valueStatus;
+
+    // This enum indicates how synchronized this entry is with storage.
+    EntrySyncStatus syncStatus = EntrySyncStatus::NOT_IN_CACHE;
   public:
     auto getValueStatus() const {
       return valueStatus;
     }
 
+    // TODO(now): We have `isDirty()`, which tells me we might also want methods for other states,
+    // but the KJ_FAIL_ASSERT is throwing me off a bit. Either we should drop isDirty() and use this
```

Comment:
The `KJ_FAIL_ASSERT` in `isDirty` is awkwardly encoding knowledge about its caller(s). It's nice to have that assertion there since it does narrow the space of possible states we need to worry about, but it does entangle this type with the rest of `ActorCache` a bit.